source/extensions/filters/network/thrift_proxy/config.cc

```diff
@@ -135,6 +148,21 @@ ConfigImpl::ConfigImpl(
       processFilter(filter);
     }
   }
+
+  if (config.has_trds()) {
+    if (config.has_route_config()) {
+      throw EnvoyException("both trds and route_config is present in ThriftProxy");
+    }
+    if (config.trds().config_source().config_source_specifier_case() ==
```

Comment:
@htuch
I just disabled the complete api_config_source case, but I can see there are AGGREGATED_GRPC and AGGREGATED_DELTA_GRPC api types which seems an other way for ads, however they are not implemented and not usable.
Should I count with these here to be future proof or they are just legacy dead end?

Reply:
These are the glorious future where we will have multiple ADS sources. For now you can either throw inside here or just mark all `api_config_source` as unsupported and throw if that is specified.

Reply:
Ok, if this is the multi ads then I added extra check for `api_type`, when it will be done no update required here.